client/service/test/src/lib.rs

```diff
@@ -161,32 +159,33 @@ where
 	{
 		let full_nodes = self.full_nodes.clone();
 		let light_nodes = self.light_nodes.clone();
-		let interval = Interval::new_interval(Duration::from_millis(100))
-			.map_err(|_| ())
-			.for_each(move |_| {
+		let future = async move {
+			let mut interval = time::interval(Duration::from_millis(100));
+
+			loop {
+				interval.tick().await;
+
 				let full_ready = full_nodes
 					.iter()
 					.all(|&(ref id, ref service, _, _)| full_predicate(*id, service));
 
 				if !full_ready {
-					return Ok(())
+					continue
 				}
 
 				let light_ready = light_nodes
 					.iter()
 					.all(|&(ref id, ref service, _)| light_predicate(*id, service));
 
-				if !light_ready {
-					Ok(())
-				} else {
-					Err(())
+				if light_ready {
+					return Ok(())
 				}
-			})
-			.timeout(MAX_WAIT_TIME);
+			}
+		};
 
-		match self.runtime.block_on(interval) {
-			Ok(()) => unreachable!("interval always fails; qed"),
-			Err(ref err) if err.is_inner() => (),
+		match self.runtime.block_on(async move { time::timeout(MAX_WAIT_TIME, future).await }) {
```

Comment:
I don't quite follow the logic but it looks like you swapped the logic such that this return `Ok(())` instead `Err(e)` on success (don't panic), is that correct?

This change makes sense to me if my comment above is correct

Reply:
Yes your understanding is correct